client/gui/CAnimation.cpp

```diff
@@ -753,6 +779,24 @@ void SDLImage::shiftPalette(int from, int howMany)
 	}
 }
 
+void SDLImage::adjustPalette(ColorShifter * shifter)
+{
+	SDL_Palette* palette = surf->format->palette;
+	for (int i = 0; i < 255; i++)
+	{
+		palette->colors[i] = shifter->shiftColor(originalPalette->colors[i]);
+	}
+}
+
+void SDLImage::resetPalette()
+{
+	SDL_Palette * pal = new SDL_Palette();

```

Comment:
`pal` in not used and also leak.

Reply:
Good point, mem leak is also a problem I was worrying. I think change them into SDL methods and use unique_ptr would resolve the problem.